File: 1.12.c
#include <stdio.h>

/*
 *	cat, with one word per line
 */

#define IN  1
#define OUT 0

int main()
{
	int c, state;

	state = OUT;
	while ((c = getchar()) != EOF)
		if(c == ' ' || c == '\n' || c == '\t') {
			if(state == IN)
				printf("\n");
			state = OUT;
		} else {
			putchar(c);
			if (state == OUT)
				state = IN;
		}
}

File: 1.12.py
#!/usr/bin/env python3

# cat, one word per line

from fileinput import input
from sys import stdout

in_word = False

for line in input():
    for c in line:
        if c in (' ', '\t', '\n'):
            if in_word:
                stdout.write('\n')
            in_word = False
        else:
            stdout.write(c)
            in_word = True
